registry/remote/repository.go

```diff
@@ -194,27 +194,43 @@ func (r *Repository) FetchReference(ctx context.Context, reference string) (ocis
 // the same base reference with the Repository r, ParseReference returns a
 // wrapped error ErrInvalidReference.
 func (r *Repository) ParseReference(reference string) (registry.Reference, error) {
-	ref, err := registry.ParseReference(reference)
-	if err != nil {
-		// reference is not a FQDN
-		if index := strings.IndexByte(reference, '@'); index != -1 {
-			// drop tag since the digest is present
-			reference = reference[index+1:]
-		}
+	var err error
+	var ref registry.Reference
+
+	if ref, err = registry.ParseReference(reference); err != nil {
```

Comment:
The original code is much simpler.
```suggestion
	ref, err := registry.ParseReference(reference)
	if err != nil {
```